src/core/state/customRPC/index.ts

```diff
@@ -31,10 +35,18 @@ export const customRPCsStore = createStore<CustomRPCsState>(
         chains: [],
         activeRpcUrl: '',
       };
-      customChain.chains.push(chain);
-      if (!customChain.activeRpcUrl)
-        customChain.activeRpcUrl = chain.rpcUrls.default.http[0];
-      set({ customChains: { ...customChains, [chain.id]: customChain } });
+      const currentRpcs = customChain.chains.map(
+        (chain) => chain.rpcUrls.default.http[0],
+      );
+      if (!currentRpcs.includes(chain.rpcUrls.default.http[0])) {
+        customChain.chains.push(chain);
+        if (!customChain.activeRpcUrl)
+          customChain.activeRpcUrl = chain.rpcUrls.default.http[0];
```

Comment:
btw that's the same as 
```ts
customChain.activeRpcUrl ||= chain.rpcUrls.default.http[0];
```
just sharing new js tricks, up to you on whats better for readability

Reply:
def. the former 😭